src/lib.rs

```diff
@@ -1 +1,452 @@
+                    eccentricity,
+                    t,
+                })
+            }
+            original::Error::NegativeBrouwerMeanMotion => SgpError::NegativeBrouwerMeanMotion,
+            original::Error::NegativeKozaiMeanMotion => SgpError::NegativeKozaiMeanMotion,
+            original::Error::NegativeSemiLatusRectum { t } => {
+                SgpError::NegativeSemiLatusRectum(NegativeSemiLatusRectum { t })
+            }
+            original::Error::Tle {
+                what,
+                line,
+                start,
+                end,
+            } => SgpError::Tle(Tle {
+                what: what.into(),
+                line: line.into(),
+                start: start.try_into().unwrap(),
+                end: end.try_into().unwrap(),
+            }),
+        }
+    }
+}
+
+impl From<original::Elements> for ElementS {
+    fn from(elements: original::Elements) -> Self {
+        let original::Elements {
+            object_name,
+            international_designator,
+            norad_id,
+            classification,
+            datetime,
+            mean_motion_dot,
+            mean_motion_ddot,
+            drag_term,
+            element_set_number,
+            inclination,
+            right_ascension,
+            eccentricity,
+            argument_of_perigee,
+            mean_anomaly,
+            mean_motion,
+            revolution_number,
+            ephemeris_type,
+        } = elements;
+        ElementS {
+            object_name,
+            international_designator,
+            norad_id,
+            classification: classification.into(),
+            datetime: datetime.to_string(),
+            mean_motion_dot,
+            mean_motion_ddot,
+            drag_term,
+            element_set_number,
+            inclination,
+            right_ascension,
+            eccentricity,
+            argument_of_perigee,
+            mean_anomaly,
+            mean_motion,
+            revolution_number,
+            ephemeris_type,
+        }
+    }
+}
+
+impl Into<original::Elements> for ElementS {
+    fn into(self) -> original::Elements {
+        let ElementS {
+            object_name,
+            international_designator,
+            norad_id,
+            classification,
+            datetime,
+            mean_motion_dot,
+            mean_motion_ddot,
+            drag_term,
+            element_set_number,
+            inclination,
+            right_ascension,
+            eccentricity,
+            argument_of_perigee,
+            mean_anomaly,
+            mean_motion,
+            revolution_number,
+            ephemeris_type,
+        } = self;
+        original::Elements {
+            object_name,
+            international_designator,
+            norad_id,
+            classification: classification.into(),
+            datetime: NaiveDateTime::from_str(&datetime).expect("Failed to parse NaiveDateTime"),
```

Comment:
I don't know how I feel about a `datetime` being stored as a `string` and potentially being passed in by the user. What are the chances this will fail and we'll blow up at runtime?

Depending on how `ElementsS` is used, we might want to consider accepting some sort of `record unix-timestamp { secs: i64, nanos: u32 }` which contains the number of seconds since 01/01/1970.